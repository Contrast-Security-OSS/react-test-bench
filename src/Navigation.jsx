import { Link, NavLink, matchPath, withRouter } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import NavDropdown from 'react-bootstrap/NavDropdown';
import { ROUTES } from './constants';

/**
 * @param {Object} props
 * @param {import('history').Location} props.location
 */
function Navigation(props) {
  return (
    <Navbar bg="light" expand="md">
      <Container>
        <Link
          // as={Navbar.Brand} throws a propType warning
          className="navbar-brand"
          to={ROUTES.INDEX}
        >
          React Test Bench
        </Link>
        <Navbar.Toggle aria-controls="nav-collapse" />
        <Navbar.Collapse id="nav-collapse">
          <Nav>
            <NavDropdown
              active={matchPath(props.location.pathname, {
                path: ROUTES.VULNERABILTIES.INDEX,
              })}
              id="nav-browser-vulns"
              title="Browser Vulnerabilities"
            >
              <NavLink
                className="dropdown-item"
                exact
                to={ROUTES.VULNERABILTIES.INDEX}
              >
                Index
              </NavLink>
              <NavLink
                className="dropdown-item"
                to={ROUTES.VULNERABILTIES.DOM_XSS}
              >
                DOM XSS
              </NavLink>
            </NavDropdown>
            <NavLink
              // as={Nav.Link} throws a propType warning
              className="nav-link"
              exact
              to={ROUTES.SERVER_VULNERABILTIES.INDEX}
            >
              Server Vulnerabilities
            </NavLink>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}

export default withRouter(Navigation);
