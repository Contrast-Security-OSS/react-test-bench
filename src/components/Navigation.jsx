import { Link, NavLink, matchPath, withRouter } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import NavDropdown from 'react-bootstrap/NavDropdown';
import { ROUTES } from '../constants';

/**
 * @param {Object} props
 * @param {import('history').Location} props.location
 * @param {Object} props.server
 * @param {string} props.server.framework
 */
function Navigation({ location, server }) {
  return (
    <Navbar bg="light" expand="md">
      <Container>
        <Link className="navbar-brand" to={ROUTES.INDEX}>
          React Test Bench
        </Link>
        <Navbar.Toggle aria-controls="nav-collapse" />
        <Navbar.Collapse id="nav-collapse">
          <Nav>
            <NavDropdown
              active={matchPath(location.pathname, {
                path: ROUTES.VULNERABILTIES.INDEX,
              })}
              id="nav-browser-vulns"
              title="Browser Vulnerabilities"
            >
              <NavLink
                className="dropdown-item"
                exact
                to={ROUTES.VULNERABILTIES.INDEX}
              >
                Index
              </NavLink>
              <NavDropdown.Divider />
              <NavLink
                className="dropdown-item"
                to={ROUTES.VULNERABILTIES.DOM_XSS}
              >
                DOM XSS
              </NavLink>
            </NavDropdown>
            {server.framework ? (
              <NavLink
                className="nav-link"
                exact
                to={ROUTES.SERVER_VULNERABILTIES.INDEX}
              >
                {server.framework} Vulnerabilities
              </NavLink>
            ) : (
              <Nav.Link disabled="true">Server Vulnerabilities</Nav.Link>
            )}
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}

export default withRouter(Navigation);
