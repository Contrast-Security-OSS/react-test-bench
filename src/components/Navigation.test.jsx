import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Navigation from './Navigation';

test('renders a navigation menu with server vulnerabilities placeholder link', () => {
  const server = {};
  const { container } = render(
    <MemoryRouter>
      <Navigation server={server} />
    </MemoryRouter>
  );
  expect(screen.getByText('Server Vulnerabilities')).toHaveClass(
    'nav-link disabled'
  );
  expect(container).toMatchSnapshot();
});

test('renders a navigation menu with an Express vulnerabilities link', () => {
  const server = { framework: 'Express' };
  const { container } = render(
    <MemoryRouter>
      <Navigation server={server} />
    </MemoryRouter>
  );
  expect(screen.getByText('Express Vulnerabilities')).toHaveClass('nav-link');
  expect(container).toMatchSnapshot();
});
