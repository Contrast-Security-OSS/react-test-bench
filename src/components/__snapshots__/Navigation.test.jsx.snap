// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a navigation menu with an Express vulnerabilities link 1`] = `
<div>
  <nav
    class="navbar navbar-expand-md navbar-light bg-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        React Test Bench
      </a>
      <button
        aria-controls="nav-collapse"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="nav-collapse"
      >
        <div
          class="navbar-nav"
        >
          <div
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              class="dropdown-toggle nav-link"
              href="#"
              id="nav-browser-vulns"
              role="button"
            >
              Browser Vulnerabilities
            </a>
          </div>
          <a
            class="nav-link"
            href="/server-vulnerabilities"
          >
            Express
             Vulnerabilities
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`renders a navigation menu with server vulnerabilities placeholder link 1`] = `
<div>
  <nav
    class="navbar navbar-expand-md navbar-light bg-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        React Test Bench
      </a>
      <button
        aria-controls="nav-collapse"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="nav-collapse"
      >
        <div
          class="navbar-nav"
        >
          <div
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              class="dropdown-toggle nav-link"
              href="#"
              id="nav-browser-vulns"
              role="button"
            >
              Browser Vulnerabilities
            </a>
          </div>
          <a
            aria-disabled="true"
            class="nav-link disabled"
            href="#"
            role="button"
            tabindex="-1"
          >
            Server Vulnerabilities
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
